### ISSUE TITLE
What is the difference between predict() and forward() of FFN

### ISSUE THREAD

**Issue by anthai0908** [mlpack/mlpack#3965]
Hi, I'm trying to use mlpack to build PPO. I'm wondering what is the difference between Predict() and Forward() of an FFN, is it about gradient tracking? Like with Predict() the gradient is not tracked?

**Comment by kumarutkarsh1248** [comment]
[Forward() implementation](https://github.com/mlpack/mlpack/blob/82f4ff83ca369479c8e90d2fa499344de9e1fbec/src/mlpack/methods/ann/ffn_impl.hpp#L310C1-L313)

The Forward() method is designed to be used inside the training loop so that the outputs of intermediate layers can be stored, which are later used in the backpropagation step.

The Predict() method, on the other hand, simply takes the input and produces the final output without storing the outputs of intermediate layers.

However, you can use either the Forward() or Predict() method to get the network output, both should produce the same result. but Forward() involves more overhead due to the extra storage.
